hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/memory/consumer/impl/id/StringIdOffHeap.java

```diff
@@ -0,0 +1,140 @@
+
+package org.apache.hugegraph.memory.consumer.impl.id;
+
+import static org.apache.hugegraph.backend.id.IdGenerator.compareType;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.hugegraph.backend.id.Id;
+import org.apache.hugegraph.backend.id.IdGenerator;
+import org.apache.hugegraph.memory.consumer.OffHeapObject;
+import org.apache.hugegraph.memory.pool.MemoryPool;
+import org.jetbrains.annotations.TestOnly;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+
+public class StringIdOffHeap extends IdGenerator.StringId implements OffHeapObject {
+
+    private ByteBuf idOffHeap;
+
+    public StringIdOffHeap(MemoryPool memoryPool, String id) {
+        super(id);
+        serializeSelfToByteBuf(memoryPool);
+        releaseOriginalVarsOnHeap();
+        memoryPool.bindMemoryConsumer(this);
+    }
+
+    public StringIdOffHeap(MemoryPool memoryPool, byte[] bytes) {
+        super(bytes);
+        serializeSelfToByteBuf(memoryPool);
+        releaseOriginalVarsOnHeap();
+        memoryPool.bindMemoryConsumer(this);
+    }
+
+    @Override
+    public Object zeroCopyReadFromByteBuf() {
+        try {
+            return new IdGenerator.StringId(idOffHeap.toString(StandardCharsets.UTF_8));
+        } finally {
+            idOffHeap.resetReaderIndex();
+        }
+    }
+
+    @Override
+    public void serializeSelfToByteBuf(MemoryPool memoryPool) {
+        byte[] stringBytes = id.getBytes((StandardCharsets.UTF_8));
+        this.idOffHeap = (ByteBuf) memoryPool.requireMemory(stringBytes.length, memoryPool);
+        this.idOffHeap.markReaderIndex();
+        this.idOffHeap.writeBytes(stringBytes);
+    }
+
+    @Override
+    public void releaseOriginalVarsOnHeap() {
+        this.id = null;
+    }
+
+    @Override
+    public List<ByteBuf> getAllMemoryBlock() {
+        return Collections.singletonList(idOffHeap);
+    }
+
+    @Override
+    public Object asObject() {
+        return this.asString();
+    }
+
+    @Override
+    public String asString() {
+        try {
+            return this.idOffHeap.toString(StandardCharsets.UTF_8);
+        } finally {
+            this.idOffHeap.resetReaderIndex();
+        }
+    }
+
+    @Override
+    public long asLong() {
+        return Long.parseLong(this.asString());
+    }
+
+    @Override
+    public byte[] asBytes() {
+        return ByteBufUtil.getBytes(this.idOffHeap);
+    }
+
+    @Override
+    public int length() {
+        return this.asString().length();
+    }
+
+    @Override
+    public int compareTo(Id other) {
+        int cmp = compareType(this, other);
+        if (cmp != 0) {
+            return cmp;
+        }
+        return this.asString().compareTo(other.asString());
+    }
+
+    @Override
+    public int hashCode() {
+        return this.idOffHeap.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof StringIdOffHeap)) {
```

Comment:
This may introduce a potential bug. If the other Object is a stringid, they are logically equal, but the original logic is changed here.